spec/formatting.md

```diff
@@ -222,6 +222,71 @@ the following steps are taken:
 
    The form that resolved _operand_ and _option_ values take is implementation-defined.
 
+   A _declaration_ binds the resolved value of an _expression_
+   to a _variable_.
+   Thus, the result of one _function_ is potentially the _operand_
+   of another _function_,
+   or the value of one of the _options_ for another function.
+   For example, in
+   ```
+   .input {$n :number minIntegerDigits=3}
+   .local $n1 = {$n :number maxFractionDigits=3}
+   ```
+   the value bound to `$n` is the
+   resolved value used as the _operand_
+   of the `:number` _function_
+   when resolving the value of the _variable_ `$n1`.
+
+   Implementations that provide a means for defining custom functions
+   SHOULD provide a means for function implementations
+   to return values that contain enough information
+   (e.g. a representation of
+   the resolved _operand_ and _option_ values
+   that the function was called with)
+   to be used as arguments to subsequent calls
+   to the function implementations.
+   For example, an implementation might define an interface that allows custom function implementation.
+   Such an interface SHOULD define an implementation-specific
+   argument type `T` and return type `U`
+   for implementations of functions
+   such that `U` can be coerced to `T` without loss of information.
+   Implementations of a _function_ SHOULD emit an 
+   _Invalid Expression_ error for _operands_ whose resolved value
+   or type is not supported.
+
+> [!NOTE]
+> The behavior of the previous example is
+> implementation-dependent. Supposing that
```

Comment:
```suggestion
> currently implementation-dependent. Supposing that
```

I disagree with this note as stated. I agree that the Tech Preview specification of the functions it has is _not sufficient_ to determine what the results of the cited example should be.

But that doesn't mean that our hands are tied from doing that by the time of the stable version in v46, which this seems to do.

Reply:
Totally agreed.